core/blockchain.go

```diff
@@ -885,7 +886,11 @@ func (self *BlockChain) InsertChain(chain types.Blocks) (int, error) {
 
 		// Create a new statedb using the parent block and report an
 		// error if it fails.
-		statedb, err := state.New(self.GetBlock(block.ParentHash()).Root(), self.chainDb)
+		if statedb == nil {
+			statedb, err = state.New(self.GetBlock(block.ParentHash()).Root(), self.chainDb)
+		} else {
+			statedb, err = statedb.New(chain[i-1].Root())
+		}
```

Comment:
Note that the call here makes it look like the state root is actually changing but that should never be the case.
After importing block i-1 the root of statedb must always be equal to `chain[i-1].Root` so in theory you don't need
to do anything here. You can just keep statedb around.